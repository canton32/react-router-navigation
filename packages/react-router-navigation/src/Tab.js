/* @flow */

import React from 'react'
import { Route } from 'react-router'
import type { TabProps } from './TypeDefinitions'
import SceneView from './SceneView'

type Props = TabProps

const Tab = (props: Props) => (
  <Route>
    {({ history }) => <SceneView {...props} type="tab" history={history} />}
  </Route>
)

export default Tab
