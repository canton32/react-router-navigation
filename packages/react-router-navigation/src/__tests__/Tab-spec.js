import React from 'react'
import { Text } from 'react-native'
import { Router } from 'react-router'
import createHistory from 'history/createMemoryHistory'
import renderer from 'react-test-renderer'
import './__mocks__'
import Tab from './../Tab'

it('<Tab /> renders correctly', () => {
  const history = createHistory({
    initialEntries: ['/1'],
  })
  const component = renderer.create(
    <Router history={history}>
      <Tab
        path="/:id"
        render={({ match: { params } }) => <Text>{params.id}</Text>}
      />
    </Router>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
