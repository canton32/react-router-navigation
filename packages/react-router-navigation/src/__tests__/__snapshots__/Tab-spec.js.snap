// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> renders correctly 1`] = `
<Text>
  1
</Text>
`;
